catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -331,11 +350,26 @@ private void addDistanceSort(SolrQuery query, String sortField, SolrQuery.ORDER
 
     protected String setSortProperty(QueryRequest request, SolrQuery query,
             SolrFilterDelegate solrFilterDelegate) {
-        SortBy sortBy = request.getQuery()
-                .getSortBy();
+
+        List<SortBy> sortBys = new ArrayList<>();
         String sortProperty = "";
+        if (request.getQuery() != null) {
+            SortBy querySortBy = request.getQuery()
+                    .getSortBy();
+            if (querySortBy != null && querySortBy.getPropertyName() != null) {
+                sortBys.add(querySortBy);
+            }
+        }
+
+        Serializable sortBySer = request.getPropertyValue(EXT_SORT_BY);
+        if (sortBySer instanceof SortBy[]) {
+            SortBy[] extSortBys = (SortBy[]) sortBySer;
+            if (extSortBys.length > 0) {
+                sortBys.addAll(Arrays.asList(extSortBys));
+            }
+        }
 
-        if (sortBy != null && sortBy.getPropertyName() != null) {
+        for (SortBy sortBy : sortBys) {
```

Comment:
❗️ Please add tests to `SolrProviderTest` for the multiple sortBys cases.  I am not sure if sorting by distance and relevance together will work since they both use `score` currently.